alibi_detect/od/_svm.py

```diff
@@ -0,0 +1,234 @@
+from typing import TYPE_CHECKING, Any, Dict, Optional, Tuple, Union
+
+import numpy as np
+
+from alibi_detect.base import (BaseDetector, FitMixin, ThresholdMixin,
+                               outlier_prediction_dict)
+from alibi_detect.exceptions import _catch_error as catch_error
+from alibi_detect.od.pytorch import SVMTorch
+from alibi_detect.od.sklearn import SVMSklearn
+from alibi_detect.utils._types import Literal
+from alibi_detect.utils.frameworks import BackendValidator
+from alibi_detect.version import __version__
+
+if TYPE_CHECKING:
+    import torch
+
+
+backends = {
+    'pytorch': SVMTorch,
+    'sklearn': SVMSklearn
+}
+
+
+class SVM(BaseDetector, ThresholdMixin, FitMixin):
+    def __init__(
+        self,
+        n_components: Optional[int] = None,
+        backend: Literal['pytorch', 'sklearn'] = 'sklearn',
+        device: Optional[Union[Literal['cuda', 'gpu', 'cpu'], 'torch.device']] = None,
+        kernel: Union['torch.nn.Module', Literal['linear', 'poly', 'rbf', 'sigmoid']] = 'rbf',
```

Comment:
The kernel definition here has become a little unwieldy and inconsistent with how we do things elsewhere. Also, although it is technically possible, users should not be specifying linear or polynomial kernels for this detector as the method relies on the kernel inducing an infinite dimensional RKHS.

I wonder if we instead encourage users to pass kernels in exactly the same way as they do for other detectors (so here we'd just have the `kernel` kwarg and not the `sigma` and `kernel_params` kwarg). We then always do our own nystrtoem approximation to produce kernalised features and only rely on sklearn for the OCSVM part?

Reply:
I think consistency across detector definitions is particularly important given we envisage the primary use-case of the detector is as a component within a large ensemble.